bundles/org.openhab.binding.panasonicbdp/src/main/java/org/openhab/binding/panasonicbdp/internal/handler/PanaBlurayHandler.java

```diff
@@ -0,0 +1,395 @@
+            return;
+        }
+
+        if (!playerKey.isBlank()) {
+            this.playerKey = playerKey;
+            authEnabled = true;
+        }
+        refreshInterval = config.refresh;
+
+        updateStatus(ThingStatus.UNKNOWN);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * Start the job to periodically get a status update from the player
+     */
+    private void startAutomaticRefresh() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            this.refreshJob = scheduler.scheduleWithFixedDelay(this::refreshPlayerStatus, 0, refreshInterval,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    /**
+     * Sends commands to the player to get status information and updates the channels
+     */
+    private void refreshPlayerStatus() {
+        final String[] playerStatusRespArr = sendCommand(REVIEW_POST_CMD, urlStr).split(CRLF);
+
+        if (playerStatusRespArr.length == 1 && playerStatusRespArr[0].isBlank()) {
+            return;
+        } else if (playerStatusRespArr.length >= 2) {
+            // CMD_REVIEW response second line, 4th group is the status:
+            // F,,,07,00,,8,,,,1,000:00,,05:10,F,FF:FF,0000,0000,0000,0000
+            final String playerStatusArr[] = playerStatusRespArr[1].split(COMMA);
+
+            if (playerStatusArr.length >= 4) {
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+                }
+
+                // update playerMode if different
+                if (!playerMode.equals(playerStatusArr[3])) {
+                    playerMode = playerStatusArr[3];
+                    final String i18nKey = STATUS_MAP.get(playerMode) != null ? STATUS_MAP.get(playerMode) : "unknown";
+                    updateState(PLAYER_STATUS, new StringType(translationProvider.getText(bundle, "status." + i18nKey,
+                            i18nKey, localeProvider.getLocale())));
+                    updateState(CONTROL, PLAY_STATUS.equals(playerMode) ? PlayPauseType.PLAY : PlayPauseType.PAUSE);
+
+                    // playback stopped, tray open, or power switched off, zero out time and chapters
+                    if (!isPlayingMode()) {
+                        updateState(TIME_ELAPSED, UnDefType.UNDEF);
+                        updateState(TIME_TOTAL, UnDefType.UNDEF);
+                        updateState(CHAPTER_CURRENT, UnDefType.UNDEF);
+                        updateState(CHAPTER_TOTAL, UnDefType.UNDEF);
+                    }
+                }
+
+                if (debounce) {
+                    updateState(POWER, OnOffType.from(!OFF_STATUS.equals(playerMode)));
+                }
+                debounce = true;
+            } else {
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.polling");
```

Comment:
```suggestion
                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/error.polling");
```

Reply:
I used this convention instead of OFFLINE to indicate that player was responding although in an unexpected way. In this state, it may still be possible to send control commands so OFFLINE seemed incorrect.